lib/atom-hack.coffee

```diff
@@ -1,49 +1,72 @@
+CP = require 'child_process'
+Path = require 'path'
+
+# This is an implementation detail of your own linter.
+lintHack = (ActiveEditor) ->
+  return new Promise (Resolve)->
+    FilePath = ActiveEditor.getPath()
+    return unless FilePath # Files that have not be saved
+    Data = []
+    Process = CP.exec('hh_client --json', {cwd: Path.dirname(FilePath)})
+    Process.stderr.on 'data', (data)-> Data.push(data.toString())
+    Process.on 'close', ->
+      try
+        Content = JSON.parse(Data.join(''))
+      catch error
+        # Ignore weird errors for now
+        # We still need to return an array so the transformer doesn't crash
+        return []
+      return Resolve([]) if Content.passed
```

Comment:
if `Content.passed` is true would `Content.errors` be an empty array? If so, this could just resolve `Content.errors` and remove the `if`